streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignorTest.java

```diff
@@ -16,8 +16,10 @@
  */
 package org.apache.kafka.streams.processor.internals;
 
+import java.util.Arrays;
```

Comment:
I think the test coverage is not sufficient with the augmented logic here, we should at least test the following code path:

1. among clients, with EAGER set we still try to honor stickiness based on user metadata (this may be covered already).
2. among clients, with COOPERATIVE we still try to honor stickiness based on owned partitions.
3. within a client, we try to reassign tasks back to consumers if the prev consumers are fixed, and load balance is not violated.
4. within a client, we if we cannot satisfy 3), we interleave.
5. within a client, upon version probing we interleave.

Reply:
Yeah I was in the middle of extending the tests when I found the version probing bug -- I'll get back to them once I've rebased